packages/plugin-ext/src/hosted/browser/hosted-plugin.ts

```diff
@@ -456,6 +470,27 @@ export class HostedPluginSupport {
         return this.pluginPathsService.getHostStoragePath(this.workspaceService.workspace, roots);
     }
 
+    protected async getHostGlobalStoragePath(): Promise<string> {
+        const configDirUri = await this.envServer.getConfigDirUri();
+        const globalStorageFolderUri = new URI(configDirUri).resolve('globalStorage').toString();
+
+        // Make sure that folder by the path exists
+        if (!await this.fileSystem.exists(globalStorageFolderUri)) {
+            await this.fileSystem.createFolder(globalStorageFolderUri);
+        }
+
+        return new Promise<string>((resolve, reject) => {
```

Comment:
(nit) sorry i don't really get why mixing async/awat and promises is good, getting rid of promises makes code so readable and short
```ts
const fsPath = await this.fileSystem.getFsPath(globalStorageFolderUri);
if (!fsPath) {
    throw new Error(`Could not resolve the FS path for URI: ${globalStorageFolderUri}`);
}
return fsPath;
```